File: lib/Device.ts
import { Date } from "./@types/index.js";

export class Device {
    public ip: string;
    public system: number;
    public date: Date;
    public time: string;
    public day: string;
    public mac: string;
    public hostname: string;
    public id: string;

    constructor() {
        this.ip = "";
        this.system = 0;
        this.date = { numeric: "", text: "" };
        this.time = "";
        this.day = "";
        this.mac = "";
        this.hostname = "";
        this.id = "";
    }
}

File: internal/find/device.go
package find

// Date holds numeric and text representations of a date.
type Date struct {
	Numeric string `json:"numeric"`
	Text    string `json:"text"`
}

// Device represents a discovered NetLinx device.
type Device struct {
	IP       string `json:"ip"`
	System   int    `json:"system"`
	Date     Date   `json:"date"`
	Time     string `json:"time"`
	Day      string `json:"day"`
	MAC      string `json:"mac"`
	Hostname string `json:"hostname"`
	ID       string `json:"id"`
}
